src/apps/activities/api/activities.py

```diff
@@ -157,6 +159,65 @@ async def applet_activities_for_subject(
         return Response(result=result)
 
 
+async def applet_activities_for_target_subject(
+    applet_id: uuid.UUID,
+    subject_id: uuid.UUID,
+    user: User = Depends(get_current_user),
+    language: str = Depends(get_language),
+    session=Depends(get_session),
+    answer_session=Depends(get_answer_session),
+) -> ResponseMulti[ActivityOrFlowWithAssignmentsPublic]:
+    applet_service = AppletService(session, user.id)
+    await applet_service.exist_by_id(applet_id)
+
+    await SubjectsService(session, user.id).exist_by_id(subject_id)
+
+    # Restrict the endpoint access to owners, managers, coordinators, and assigned reviewers
+    await CheckAccessService(session, user.id).check_subject_subject_access(applet_id, subject_id)
+
+    assignments = await ActivityAssignmentService(session).get_all_with_subject_entities(
+        applet_id, QueryParams(filters={"target_subject_id": subject_id})
+    )
+
+    activity_and_flow_ids = await AnswerService(
+        session, user.id, answer_session
+    ).get_activity_and_flow_ids_by_target_subject(subject_id)
+
+    activities_and_flows = await ActivityService(session, user.id).get_activity_and_flow_basic_info_by_ids_or_auto(
+        applet_id, activity_and_flow_ids, language
+    )
+
+    result = []
+    for activity_or_flow in activities_and_flows:
+        activity_or_flow_assignments = [
+            assignment
+            for assignment in assignments
+            if assignment.activity_id == activity_or_flow.id or assignment.activity_flow_id == activity_or_flow.id
+        ]
+
+        if len(activity_or_flow_assignments) > 0:
+            if activity_or_flow.is_hidden:
```

Comment:
`is_hidden` check should take priority over assignment status – i.e., it should return `hidden` if there are 0 manual assignments, but `auto_assign` is true. So I'd do:

```py
if activity_of_flow.is_hidden:
    activity_or_flow.status = ActivityOrFlowStatusEnum.HIDDEN
elif activity_or_flow.auto_assign or activity_or_flow_assignments > 0:
    activity_or_flow.status = ActivityOrFlowStatusEnum.ACTIVE
else:
    activity_or_flow.status = ActivityOrFlowStatusEnum.INACTIVE

```

Reply:
Gotcha. I updated the logic